Add a product vision and guiding principles document. (#15022)
nvaccess/nvda: product_vision.md

## Product Vision
A world where blind and vision impaired people can independently and fully interact with the Windows Operating System and popular third party applications, enabling them to contribute to society, no matter their language, location or economic status.

> Blind and vision impaired people are empowered to control their own destiny and ensure this vision, through being able to significantly contribute to the technology that makes this possible.

Reviewer: I think that under-represented people like deaf and blind should be mentioned too.

Author: I agree we must consider deaf-blind users always. I welcome any ideas on how that could be worded.
I would suggest though that The vision statement should continue to target just one group. So far it is "blind and vision impaired people", which of course is a superset containing "deaf-blind". So Deaf-blind people are certainly not excluded here.
* Should we make the group more explicit and say "blind, vision impaired, and deaf-blind"? Sounds very clunky and confusing to me.
* Can we extend "language, location and economic status" to include "abilities" or "other sensory abilities"?
* Should we add a third sentence talking explicitly about ensuring that blind and vision impaired people of all abilities are included? 
* What about motor and cognitive abilities? 
The vision statement can certainly be a perfect world we strive towards, but it also should be attainable, and not so large that we could never manage it.  
How ever we improve this statement, we will also need to ensure it is backed up by the guiding principles. E.g. if we do acknowledge motor abilities, then we should add voice input to the multimodal goal. Which although is the right thing to do, we also commit the project to ensuring that voice input is actually available.
Thoughts anyone?

Reviewer: > • Should we make the group more explicit and say "blind, vision impaired, and deaf-blind"? Sounds Sounds very clunky and confusing to me.

I agree: this is confusing and redundant.

> Can we extend "language, location and economic status" to include "abilities" or "other sensory abilities"?

I think we can extend "language, location and economic status", but I prefer sensory characteristics instead of abilities, since characteristic sounds more qualitative instead of something measurable in terms of numerical values. Anyway I'm not a native English speaker.

> What about motor and cognitive abilities?

I thought about this too, and even mental or psichological states, not necessarily, but also mental illness. In Spain usually it's distinguished between cognitive and psichosocial (previously named mental) disabilities. I understand that NVDA itself may not contain specific code for mental abilities, but the code ofconduct aims to decrease stress, so I think that maybe mentioned in this principles.

Author: I have tried to address your points in my last commit. However, I have chosen to keep with the word 'abilities', as I believe that is still the most suitable word in English.
I removed language, location and economic status from the first sentence, and moved it to its own third sentence, extending it to include sensory, physical, cognitive, and mental abilities. I also added a specific example to the multimodal section of the guiding principles, which specifically mentions deaf-blind users being able to use NVDA with Braille alone.

Reviewer: For me is good. Thanks.

## Product description
NVDA (NonVisual Desktop Access) is software that allows blind and vision impaired people to independently interact with the Windows Operating System and popular third party applications by outputting content in either synthetic speech or via a refreshable Braille Display.

> Developed by NV Access in partnership with a dedicated open source community, NVDA enables blind and vision impaired people to increase their chances at education and employment, and to make a contribution to society.

Reviewer 2: This is perhaps more of a political question - but I am interested to know thoughts on rephrasing "make a contribution to society" to "participate within society". 
The phrase "contribute to society" is also used in the first sentence of the product vision.

The difference being an expectation of contributing / performing work, which might not be suitable to all users.
Lowering barriers for employment and education is great - but I've also heard user stories about NVDA aiding participation in society in other ways, e.g. social platforms.

Maybe this can be a separate added sentence rather than rephrasing this one, mentioning that NVDA aims to enable users to participate in society, outside education and employment.

Reviewer: I strongly agree with @seanbudd. Participation is related to something much more active than productive contribution, who may not be possible for everyone due to expectations and considerations about what productivity it. Contribution sounds like to give something to a given social structire, and participation seems more precise meaning action,not just accepting and helping.

Author: Some good points here.
As context, ever since the beginning Jamie and I have always said "contribute to society" as we believe that everyone has a right to contribute to society, though some may require some extra tools to do so.
Also note that the statement says "allow", I.e. we are allowing people to contribute if they wish, not forcing or expecting them to.
"Participate" may be a more acceptable word. What are your thoughts @jcsteh ? I think both have their own valid need. "participate in and or contribute to society" is what I would want, but perhaps "participate" covers both enough.

Reviewer 3: This is an interesting question. I've always used the word "contribute" because I strongly believe that everyone has a right but also a responsibility to contribute to society in some way. Historically, even where people with disabilities might not have been excluded from participating in certain aspects of society, there were more often excluded from contributing because they didn't have the right tools or because they weren't given the right opportunities. I believe this also created a situation where this norm became so firmly established that even where opportunities and tools became available, some people with disabilities didn't contribute (and perhaps didn't even feel they should have to) even though they could have. Given the right tools and opportunities, I don't think anyone should be excluded from this responsibility or the rights that come with it. Reflecting on this, I do realise this stance comes to some extent from a position of privilege. Just because I have opportunities doesn't mean that everyone does.

Also, it's worth noting that contributing to society doesn't necessarily mean traditional contribution. You don't have to work a paid job and pay taxes to contribute. Perhaps you volunteer where it's needed most, etc.

All of that said, I think it's reasonable to say "participate in and contribute to society". I don't think I'd want to drop contribute to - I still think this is important for all the reasons outlined above - but adding "participate" makes it clear that we think both are important. As @MichaelDCurran noted, "enables" also makes it clear that we want to enable these things, but that we can't force or require our users to do them.

    * Documentation is provided to ensure code can be understood and developers can get up and running with ease.
    * The vast majority of contributers to the NvDA project are blind or vision impaired, and are users of the product themselves.
3. Multilingual: Ensure that NVDA can be used by people no matter what language they speak:
    * NVDA provides mechanisms for volunteers to translate user interface messages and documentation.
    * Release processes provide time and reminders for volunteers to contribute translations to NVDA. 
4. Stability: Ensure NVDA is stable, minimizes freezes, and avoids drastic updates that could disrupt user experience:
    * NVDA strives to support any breaking changes in Windows or supported popular third party applications before they reach the user.
    * NVDA updates should not drastically change or remove pre-existing features or defined behaviour without clearly warning the user first. 
    * NVDA's code should do its best to recover from application freezes, allowing the user to still interact with the rest of the Operating System.

> 5. Quality: Ensure NvDA's code is high-quality, safe, and that bugs are caught before reaching a release:

Reviewer 2: Capitalise v in NVDA

```suggestion
5. Quality: Ensure NVDA's code is high-quality, safe, and that bugs are caught before reaching a release:
```

    * Automated unit and system tests decrease the possibility of unwanted changes in behaviour.
    * Alpha and beta builds are provided for early testing by a large base of dedicated community testers
6. Security: Ensure that NVDA does not place a user at risk of security attacks:
    * NVDA prioritizes user security, striving to eliminate risks of remote control attacks or data leaks.
    * A strict process exists to follow a security report from implementation to rollout.
    * Critical security fixes are minimized in size and complexity to focus on the exact issue reported, which lessens the chance of other security or instabilities accidentally being introduced. 
7. Efficient: Ensure responsiveness to user input, but also energy efficiency where possible:
    * Make use of newer more efficient APIs where possible, at the same time falling back to older APIs to ensure functionality on older systems. 
    * Periodically remove technical debt and refactor code to make code more efficient and understandable, making sure to maintain compatibility with existing scenarios and OS / application versions still supported.

> 8. Innovative: Strive to push the boundaries of what is possible and to avoide stagnation for both NvDA and its compeditors:

Reviewer 2: Fix various typos

```suggestion
8. Innovative: Strive to push the boundaries of what is possible and to avoid stagnation for both NVDA and its competitors:
```

    * Prototype new ideas and make them available through try builds.
    * Promote the sharing and healthy debate of ideas on developer email lists and Github Discussions.
9. Compliance: Ensure app / web developers can rely upon NVDA to make best use of the accessibility they implement:
    * NVDA strives to comply with all application and web accessibility standards where possible.
    * Where there is disagreement or inability to comply with an accessibility standard, NVDA maintainers  clearly and respectfully argue the case, providing practical alternative solutions.

>     * NVDA maintainers partispate in discussions with standards groups.

Reviewer 2: Fix typo

```suggestion
    * NVDA maintainers participate in discussions with standards groups.
```

    * NVDA delivers output through both speech synthesis and braille displays.
    * To cater to the diversity of user preferences, NVDA strives to make all its features accessible through all input and output modalities. 
12. Compatibility: Ensure that NVDA works with a wide range of software and hardware to provide maximum accessibility and to ensure users are not left behind in today's rapidly changing technology:
    * NVDA is regularly updated to be compatible with the latest versions of Windows and popular third-party applications.
    * NVDA also maintains compatibility with older versions of Windows and third-party applications that are still officially supported by their respective companies, providing accessibility solutions for users on older systems.
    * NVDA strives to support all common speech synthesizers and refreshable Braille displays that exist today.
13. Configurable: Ensure the user can configure NVDA to suit their own particular preferences and the task at hand:
    * NVDA contains many options for configuring what information is reported and how it is presented.

>     * NvDA allows the user to configure attributes of the output such as  the voice rate and pitch, punctuation, and what Braille standard is used.

Reviewer 2: Fix typos

```suggestion
    * NVDA allows the user to configure attributes of the output such as the voice rate and pitch, punctuation, and what Braille standard is used.
```

    * NVDA allows the user to create configuration profiles, so that the user can easily switch configurations depending on the particular task they are performing.
14. Extendable: Ensure that NVDA can be easily enhanced and customized to meet specific user needs and requirements:

>     * NVDA allows the creation and integration of community-created add-ons, providing support for extra speech synthesizers, refreshable Braille displays and more third party applications. 

Reviewer 2: Should this be hyphenated?

```suggestion
    * NVDA allows the creation and integration of community created add-ons, providing support for extra speech synthesizers, refreshable Braille displays and more third party applications. 
```

Reviewer: I think that vision providers are missing here.

>     * NVDA allows the user to browse for, download and install available add-ons from within NVDA itself.

Reviewer 2: Is "for" redundant here?

```suggestion
    * NVDA allows the user to browse, download and install available add-ons from within NVDA itself.
```

Reviewer: Should be added the expression add-ons "available in different languages"?

    * NVDA provides freedom for add-ons to find innovative solutions by allowing them broad access to NVDA's internals.

>     * NVDA ensures add-ons do not break by postponing major API changes to once per year, announcing these changes many months ahead of time.

Reviewer 2: Should this refer to breakages not changes? We regular make major API changes - e.g. by introducing new features.

```suggestion
    * NVDA ensures add-ons do not break by postponing API breakages to once per year, announcing these changes many months ahead of time.
```

Reviewer: I agree with @seanbudd
